### ISSUE TITLE
abi: `eth_abi_mpint` should encode `mpz`s as integers, not bytes.

### ISSUE THREAD

**Issue by mhw0** [mhw0/libethc#20]

**Comment by DerXanRam** [comment]
> Hey @DerXanRam. Can you please pull the `dev` branch and test if it fixes the issue? [77526b3](https://github.com/mhw0/libethc/commit/77526b38b124fe1728b98d8a2645856037fee1d5)

ok wait...

**Comment by DerXanRam** [comment]
please give me 1 or 2 hour... i'm on way

**Comment by mhw0** [comment]
Cool. I'll close the issue as soon as we merge the `dev` branch. Thanks for testing.

**Comment by DerXanRam** [comment]
> Cool. I'll close the issue as soon as we merge the `dev` branch. Thanks for testing.

Hello brother...but on some numbers it generates false numbers. look at this number

```
char is 11531912076352190230331
ABI is d06ca61f00000000000000000000000000000000000000000000000271256dc6c0e9a12f00000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000002000000000000000000000000ade00c28244d5ce17d72e40330b1c318cd12b7c3000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2
```

**Comment by mhw0** [comment]
You're right. I was assuming that the number of bytes is always even. Here's the fix: b5a77855b59c7c984e9bafd99cf8a0c6bf68f0d3

**Comment by DerXanRam** [comment]
Does the issue solved in the encoding solved also for decoding? I say this because i see some numerical instability when i decode large numbers like 24 digits,

**Comment by mhw0** [comment]
> Does the issue solved in the encoding also solved for decoding? I say this because i see some numerical instability when i decode large numbers like 24 digits,

It should give the correct result on decode.

**Comment by DerXanRam** [comment]
> > Does the issue solved in the encoding also solved for decoding? I say this because i see some numerical instability when i decode large numbers like 24 digits,
> 
> It should give the correct result on decode.

sorry ma bad... i found my problem... it woeks :+1:

**Comment by mhw0** [comment]
Closing the issue. Published in `v1.1.1`
